apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c

```diff
@@ -75,241 +64,755 @@ static void clean_all_data(int fd, char *buffer)
 }
 
 /**
-* @testcase         audio_pcm_open_close_tc_p
+* @testcase         itc_audio_pcm_open_close_p
 * @brief            open and initlaize pcm structure
 * @scenario         Open pcm and check its validation
 * @apicovered       pcm_open, pcm_close
 * @precondition     NA
 * @postcondition    NA
 */
-static void itc_audio_pcm_open_close_tc_p(void)
+static void itc_audio_pcm_open_close_p(void)
 {
+	g_pcm = NULL;
 	g_pcm = pcm_open(0, 0, PCM_IN, NULL);
+	TC_ASSERT_NEQ("pcm_open", g_pcm, NULL);
 	TC_ASSERT_GT("pcm_open", pcm_get_file_descriptor(g_pcm), 0);
+
 	TC_ASSERT_EQ("pcm_close", pcm_close(g_pcm), 0);
 	TC_SUCCESS_RESULT();
 }
 
 /**
-* @testcase         audio_pcm_is_ready_p
+* @testcase         itc_audio_pcm_open_p_with_different_flag
+* @brief            open and initlaize pcm structure
+* @scenario         Open pcm and check its validation
+* @apicovered       pcm_open, pcm_close
+* @precondition     NA
+* @postcondition    NA
+*/
+static void itc_audio_pcm_open_p_with_different_flag(void)
+{
+	int index = 0;
+	int count = 0;
+	int ret = 0;
+	bool check = true;
+	unsigned int flag_arr[] = { PCM_IN, PCM_OUT, PCM_MMAP, PCM_NOIRQ, PCM_MONOTONIC };
+	const char *ptr_flag[] = { "PCM_IN", "PCM_OUT", "PCM_MMAP", "PCM_NOIRQ", "PCM_MONOTONIC" };
+	count = sizeof(flag_arr) / sizeof(flag_arr[0]);
+
+	for (index = 0; index < count; index++) {
+		g_pcm = NULL;
+		g_pcm = pcm_open(0, 0, flag_arr[index], NULL);
+		if (g_pcm == NULL || pcm_get_file_descriptor(g_pcm) <= 0) {
+			printf("\nitc_audio_pcm_open_p_with_different_flag: pcm_open FAIL for flag : %s\n", ptr_flag[index]);
+			check = false;
+			continue;
+		}
+		ret = pcm_close(g_pcm);
+		if (ret != 0) {
+			printf("\nitc_audio_pcm_open_p_with_different_flag: pcm_close FAIL for flag : %s\n", ptr_flag[index]);
+			check = false;
+		}
+	}
+
+	TC_ASSERT_EQ("itc_audio_pcm_open_p_with_different_flag", check, true);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         itc_audio_pcm_open_p_reopen
+* @brief            open and initlaize pcm structure
+* @scenario         Open pcm and check its validation
+* @apicovered       pcm_open, pcm_close
+* @precondition     NA
+* @postcondition    NA
+*/
+static void itc_audio_pcm_open_p_reopen(void)
+{
+	struct pcm *ptr_pcm = NULL;
+	g_pcm = NULL;
+	g_pcm = pcm_open(0, 0, PCM_IN, NULL);
+	TC_ASSERT_NEQ("pcm_open", g_pcm, NULL);
+	TC_ASSERT_GT("pcm_open", pcm_get_file_descriptor(g_pcm), 0);
+	// re-open pcm stream
+	ptr_pcm = pcm_open(0, 0, PCM_IN, NULL);
+	TC_ASSERT_NEQ("pcm_open", ptr_pcm, NULL);
```

Comment:
ptr_pcm will allocated, so it is not null, you should check validation with pcm_get_file_descriptor.

Reply:
ptr_pcm is initiallised to NULL before. pcm_open on Failure can return NULL. Later ptr_pcm is also validated with pcm_get_file_descriptor